src/main/scala/org/apache/spark/sql/pulsar/PulsarMetadataReader.scala

```diff
@@ -227,7 +237,8 @@ private[pulsar] case class PulsarMetadataReader(
   }
 
   private def getTopics(): Seq[String] = {
-    topics = caseInsensitiveParameters.find(x => TOPIC_OPTION_KEYS.contains(x._1)).get match {
+    topics = caseInsensitiveParameters
+      .filterKeys(x => TopicOptionKeys.contains(x)).toSeq.head match {
```

Comment:
This modification is not good for scala style. Use `find` is better.

Reply:
Two problems for the `find`: 

`Usage of get on optional type.`

`Found unnamed result or parameter`

If you are interested, here is the detailed link: This is the code style check result link: https://app.codacy.com/gh/streamnative/pulsar-spark/pullRequest?prid=8345560

Reply:
I have read the style issues. The first one is easy to fix by using `flatmap`. The second issue would be fixed by using `caseInsensitiveParameters.find((key, _)) => TopicOptionKeys.contains(key))`

The `head` method would return `null` if no keys found in `TopicOptionKeys`. This would cause NP issues.

Reply:
I'm okay to use the `find` one here.

And just FYI, It's actually guaranteed that there'll be 1 and only 1 element in the result seq:
https://github.com/streamnative/pulsar-spark/blob/master/src/main/scala/org/apache/spark/sql/pulsar/PulsarProvider.scala#L382